RealmSwift/Realm.swift

```diff
@@ -1217,13 +1218,18 @@ extension Realm {
             }
         case .always:
             rlmRealm = try await withCheckedThrowingContinuation { continuation in
-                RLMRealm.asyncOpen(with: configuration.rlmConfiguration, callbackQueue: .main) { (realm, error) in
+                let rlmTask = RLMRealm.asyncOpen(with: configuration.rlmConfiguration, callbackQueue: .main) { (realm, error) in
```

Comment:
Hey @dianaafanador3 thanks for the PR. Looking good.

What was your reasoning for fixing the callbackQueue to `.main` rather than letting an implementer pass in their preffered queue?

Reply:
Because you can only run `let realm = try await Realm()` at the moment, in a @MainActor. 
We are still discussing if this is the bet API for this, but we have a project in progress which should give better support for actor-confined Realms and offer a better experience with Swift concurrency in general, which should include exposing recurring callbacks as `AsyncSequences` for sync progress notification.

Reply:
That's great, thanks @dianaafanador3